src/mam4xx/utils.hpp

```diff
@@ -161,71 +172,145 @@ void transfer_work_arrays_to_prognostics(const Real q[gas_pcnst()],
   }
 }
 
-// converts the quantities in the work arrays q and qqcw from mass/number
-// mixing ratios to volume/number mixing ratios
+#undef DECLARE_PROG_TRANSFER_CONSTANTS
+
+// Given an AerosolState with views for dry aerosol quantities, creates a
+// interstitial aerosols 1D view (state_q) for the column with the given index.
+// This object can be provided to mam4xx for the column.
+
+// MUST FIXME: address James comments about making the code better.
 KOKKOS_INLINE_FUNCTION
-void convert_work_arrays_to_vmr(const Real q[gas_pcnst()],
-                                const Real qqcw[gas_pcnst()],
-                                Real vmr[gas_pcnst()],
-                                Real vmrcw[gas_pcnst()]) {
-  DECLARE_PROG_TRANSFER_CONSTANTS
+void state_q_from_progs_at_one_lev(const mam4::Prognostics &progs,
```

Comment:
What about something like `extract_stateq_from_prognostics`? To me, the inclusion of `klev` and the documentation can make it clear that the extraction happens only for that level. The inverse operation (below) could be called something like `inject_stateq_to_prognostics`, too.